### PR TITLE
Added Scala language support and hello world example

### PR DESCRIPTION
**Pull request by anfelbar**
# About

Adding Scala support and hello world example. 
Run with `bazel run //scala`
Also fixed version for ubuntu iso (it was failing because of that)

Progress https://github.com/EngFlow/example/issues/36

### CODE REVIEW

**Comment by anfelbar** [comment]
Github actions are failed for macos, I will handle that in another PR (https://github.com/EngFlow/example/issues/38)
This is ready for merging.

File: WORKSPACE [hunk 1]
```diff
@@ -13,11 +15,11 @@ http_file(
 )
 
 http_file(
-    name = "ubuntu_20.04_1.3GB",
-    sha256 = "28ccdb56450e643bad03bb7bcf7507ce3d8d90e8bf09e38f6bd9ac298a98eaad",
+    name = "ubuntu_20.05_1.3GB",
```
Comment (jayconrod) [inline WORKSPACE:18]: The old name didn't have the patch version number, so let's keep it as it was.

```suggestion
    name = "ubuntu_20.04_1.3GB",
```

File: WORKSPACE [hunk 2]
```diff
@@ -122,3 +124,28 @@ maven_install(
 load("@maven//:compat.bzl", "compat_repositories")
 
 compat_repositories()
+
+
+rules_scala_version = "20220201"
```
Comment (jayconrod) [inline WORKSPACE:129]: Rename to `RULES_SCALA_VERSION`.

Conventionally in Starlark, global variable names are all caps.

File: scala/BUILD [hunk 3]
```diff
@@ -0,0 +1,7 @@
+load("@io_bazel_rules_scala//scala:scala.bzl", "scala_binary")
+
+scala_binary(
+    name = "scala",
+    srcs = glob(["com/hello/Main.scala"]),
```
Comment (jayconrod) [inline scala/BUILD:5]:
```suggestion
    srcs = glob(["*.scala"]),
```

File: scala/com/hello/Main.scala [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.hello;
```
Comment (jayconrod) [inline scala/com/hello/Main.scala:1]: To keep this parallel to the Java example, let's rename this to `com.engflow.example` and move to that directory. Also add a test.

File: scala/BUILD (hunk 3 above)
Comment (jayconrod) [inline scala/BUILD:5]:
```suggestion
    srcs = ["Main.scala"],
```

**Comment by anfelbar** [comment]
It now has same structure and tests cases as java.

**Review by ulfjack — APPROVED** [review]

**Pull request merged**